In [3]:
import tensorflow as tf
#from os import path, getcwd, chdir
path = "/content/mnist.npz"

In [4]:
tf.__version__

'2.2.0'

In [27]:

def train_mnist():
  class myCallback(tf.keras.callbacks.Callback):
    def on_epoch_end(self, epoch, logs={}):
      if(logs.get('accuracy')>0.99):
        print("\nReached 99% accuracy so cancelling training!")
        self.model.stop_training = True
     
  mnist = tf.keras.datasets.mnist
  callbacks = myCallback()
  (x_train, y_train),(x_test, y_test) = mnist.load_data(path=path)
  x_train, x_test = x_train / 255.0, x_test / 255.0
     
  model = tf.keras.models.Sequential([
                                        tf.keras.layers.Flatten(input_shape=(28, 28,)),
                                        tf.keras.layers.Dense(512, activation=tf.nn.relu),
                                        # tf.keras.layers.Dense(256, activation=tf.nn.relu),
                                        tf.keras.layers.Dense(10, activation=tf.nn.softmax)])

  model.compile(optimizer='adam',
                  loss='sparse_categorical_crossentropy',
                  metrics=['accuracy'])
    # model fitting
  history = model.fit(
        x_train,y_train, epochs = 10, callbacks = [callbacks], batch_size = 32
        )
    # model fitting
  eval1 = model.evaluate(x_test,y_test)
  return history.epoch, history.history['accuracy'][-1], eval1

In [28]:
train_mnist()

Epoch 1/10
1875/1875 [==============================] - 10s 6ms/step - loss: 0.1999 - accuracy: 0.9413
Epoch 2/10
1875/1875 [==============================] - 10s 5ms/step - loss: 0.0810 - accuracy: 0.9747
Epoch 3/10
1875/1875 [==============================] - 10s 6ms/step - loss: 0.0527 - accuracy: 0.9832
Epoch 4/10
1875/1875 [==============================] - 10s 5ms/step - loss: 0.0362 - accuracy: 0.9880
Epoch 5/10
1866/1875 [============================>.] - ETA: 0s - loss: 0.0272 - accuracy: 0.9910
Reached 99% accuracy so cancelling training!
313/313 [==============================] - 1s 2ms/step - loss: 0.0768 - accuracy: 0.9771


([0, 1, 2, 3, 4], 0.9909666776657104, [0.0768001526594162, 0.9771000146865845])

Our model gives a 99% Training accuracy and 97% Validation accuracy.
